src/setup-elp.js

```diff
@@ -1,9 +1,159 @@
 const core = require('@actions/core')
+const exec = require('@actions/exec')
+const toolCache = require('@actions/tool-cache')
+const fs = require('fs')
+const path = require('path')
 
-try {
-  const input1 = core.getInput('input-1')
-  console.log(`input-1 is ${input1}`)
-  core.setOutput('output-1', input1)
-} catch (error) {
+async function main() {
+  prepareLocal()
+  const elpVersion = getRequiredInput('elp-version')
+  assertArchsPlatforms()
+  installElp(elpVersion)
+}
+
+main().catch((error) => {
   core.setFailed(error.message)
+})
+
+async function installElp(elpVersion) {
+  const toolName = 'elp'
+  let cachePath = toolCache.find(toolName, elpVersion)
+
+  if (cachePath === '') {
+    core.debug(`ELP ${elpVersion} is not cached`)
+    const elpTarGzFile0 = await elpTarGzFile()
+    const elpTarGz = `https://github.com/WhatsApp/erlang-language-platform/releases/download/${elpVersion}/${elpTarGzFile0}`
+    core.debug(`ELP download URL is ${elpTarGz}`)
+    const file = await toolCache.downloadTool(elpTarGz)
+    const dest = undefined
+    const flags = ['zx']
+    const targetDir = await toolCache.extractTar(file, dest, flags)
+    const target = 'elp'
+    cachePath = await toolCache.cacheFile(file, target, toolName, elpVersion)
+  } else {
+    core.debug(`ELP ${elpVersion} is cached`)
+  }
+
+  const runnerToolPath = path.join(process.env.RUNNER_TEMP, '.setup-elp', 'elp')
+  fs.cpSync(cachePath, runnerToolPath, { recursive: true })
+  fs.chmodSync(path.join(runnerToolPath, 'elp'), 0o755)
```

Comment:
Not sure why this is required. Doing a regular download (`wget` + `tar zx`) fetches the proper attributes.